src/Task.ts

```diff
@@ -84,11 +84,46 @@ export class Task {
 
     public static readonly dateFormat = 'YYYY-MM-DD';
 
+    // Matches indentation before a list marker (including > for potentially nested blockquotes or Obsidian callouts)
+    public static readonly indentationRegex = /^([\s\t>]*)/;
+
+    // Matches but does not save - or * list markers.
+    public static readonly listMarkerRegex = /[-*]/;
+
+    // Matches a checkbox and saves the status character inside
+    public static readonly checkboxRegex = /\[(.)\]/;
+
+    // Matches the rest of the task after the checkbox.
+    public static readonly afterCheckboxRegex = / *(.*)/u;
+
     // Main regex for parsing a line. It matches the following:
-    // - Indentation (including > for potentially nested blockquotes or Obsidian callouts)
+    // - Indentation
     // - Status character
     // - Rest of task after checkbox markdown
-    public static readonly taskRegex = /^([\s\t>]*)[-*] +\[(.)\] *(.*)/u;
+    public static readonly taskRegex = new RegExp(
+        this.indentationRegex.source +
+            this.listMarkerRegex.source +
+            ' +' +
+            this.checkboxRegex.source +
+            this.afterCheckboxRegex.source,
+        'u',
+    );
+
+    // Used in CreateOrEdit.ts to parse indentation and status if present
+    public static readonly nonTaskRegex = new RegExp(
+        this.indentationRegex.source +
+            this.listMarkerRegex.source +
+            '? *(' +
+            this.checkboxRegex.source +
+            ')?' +
+            this.afterCheckboxRegex.source,
+        'u',
+    );
+
+    // Used in ToggleDone.ts to detect a list item that can get a checkbox added to it.
+    public static readonly listItemRegex = new RegExp(
+        this.indentationRegex.source + '(' + this.listMarkerRegex.source + ')',
+    );
```

Comment:
Thinking out loud - and not for changing in this PR... but seeing the comments that list where the regexes are used, it makes me think that there is a rather more fundamental issue with the code, which that there is scope for better abstraction in some of the core classes.

But this change is definitely a step in a good direction of removing some duplication of key text patterns.

Reply:
I am slightly thinking that comments that name where a regex is used are likely to become out of date over time, and it would be better to say what the intention of the regex was (although ideally it would be clear from the name).

IDEs are for navigating to where code is used, and such comments naming locations of use very often become out of date over time. And then misleading comments are a definite smell.

So for the avoidance of doubt, the suggestion is - for wherever file names are given in new comments:

```diff
- // Used in ToggleDone.ts to detect a list item that can get a checkbox added to it.
+ // Used to detect a list item that can get a checkbox added to it.
```